processing/src/main/java/io/druid/segment/data/VSizeLongSerde.java

```diff
@@ -0,0 +1,634 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.metamx.common.IAE;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.BufferOverflowException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * Currently only support big endian
+ * <p>
+ * An empty 4 bytes is written upon closing to avoid index out of bound exception for deserializers that shift bytes
+ */
+public class VSizeLongSerde
+{
+
+  public static final int SUPPORTED_SIZE[] = {1, 2, 4, 8, 12, 16, 20, 24, 32, 40, 48, 56, 64};
```

Comment:
I followed the same supported size for lucene, I believe the rational is that bit sizes that are not multiples of 8 takes more operations to serialize/deserialize, so performance is worse. The larger values not divisible by 8 doesn't offer a significant size reduction to next multiple of 8 compare to smaller values. For instance, 12 offers 25% size reduction to 16, while 36 only has 10% size reduction to 40, so it's not worth the performance difference.